src/librustc_mir/transform/promote_consts.rs

```diff
@@ -266,8 +277,10 @@ impl<'a, 'tcx> Promoter<'a, 'tcx> {
         });
 
         // Inject the Rvalue or Call into the promoted MIR.
+        let dest = Lvalue::Temp(new_temp);
+        self.push(StatementKind::StorageLive(dest.clone()), source_info.span);
```

Comment:
@eddyb 

> Constants can borrow things that outlive the evaluation of the constant, so in a sense, this would be valid...We could not have any Storage{Live,Dead} in constants, that would be fine with me, for now.

Can you clarify to me what you mean by this? Are you talking about borrowing other constants? Maybe an example?